### PR TITLE
View map

### PR DESCRIPTION
**Pull request by afmicc**
#### Trello references
* [As a user I should be able to allow the browser to detect my location, so the map can be positioned on it and find people with similar interests](https://trello.com/c/2lCptX0w)
* [As a user I should be able to see a map so I can easily identify targets in the future](https://trello.com/c/iilG4jVa)
* [As a user I should be able to see all my targets in the map so I can easily identify them](https://trello.com/c/J9nByHzi)

#### Description

* Add headers to request only to ones which the destiny is the API
* Show current location
* Show user targets over the map
* Add map section in existing pages

Screen Shot 2020-03-25 at 1 04 47 PM

### CODE REVIEW

File: src/components/TargetsMap/index.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+const zoom = 15;
+
+const TargetsMap = () => {
+  const { latitude, longitude } = usePosition(true);
+  const targets = useSelector(({ target: { values } }) => values);
+  const topics = useSelector(({ topic: { values } }) => values);
+
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(getTargets(1));
```
Comment (peter7z) [inline src/components/TargetsMap/index.js]: What's this `1`?

**Review by peter7z — APPROVED** [review]

File: src/components/TargetsMap/index.js (hunk 1 above)
Comment (afmicc) [reply src/components/TargetsMap/index.js]: According to the api doc, it's the page number but it seems not to be working.

Comment (afmicc) [reply src/components/TargetsMap/index.js]: Maybe, it's a good idea to create a constant with that. wdyt?

File: src/components/TargetsMap/Indicator/Indicator.scss [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  flex-direction: column;
+}
+
+.map__indicator__circle {
+  align-items: center;
+  display: flex;
+  justify-content: center;
+  transform: translate(0, -50%);
+  width: 2rem;
+}
```
Comment (manema) [inline src/components/TargetsMap/Indicator/Indicator.scss:15]: You can write lines 3 - 15 in this way
```suggestion
.map__indicator {
  align-items: center;
  display: flex;
  flex-direction: column;
  
    &__circle {
      align-items: center;
      display: flex;
      justify-content: center;
      transform: translate(0, -50%);
      width: 2rem;
   }
}
```

File: src/components/TargetsMap/Indicator/index.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { string, bool } from 'prop-types';
+
+import './Indicator.scss';
+
+const Indicator = ({ icon, label, title, background }) => (
+  <div className="map__indicator">
+    <div className={`map__indicator__circle ${background && 'map__indicator__circle--background'}`}>
```
Comment (manema) [inline src/components/TargetsMap/Indicator/index.js:8]: Take a look at classnames to manages classNames

File: src/components/TargetsMap/TargetIndicator/index.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  return <Indicator title={title} icon={icon} label={label} background />;
+};
+
+TargetIndicator.propTypes = {
+  target: shape({
+    title: string.isRequired,
+    topicId: number
+  }),
+  topics: arrayOf(
+    shape({
```
Comment (manema) [inline src/components/TargetsMap/TargetIndicator/index.js]: I guess the first shape is not necessary. Isn't it?

File: src/components/TargetsMap/index.js (hunk 1 above)
Comment (manema) [reply src/components/TargetsMap/index.js]: Agree. It must have a semantic value.

File: src/redux/reducers/topicReducer.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  error: undefined
+};
+
+export default createReducer(intialState, {
+  [getTopicsSuccess]: (state, { payload: { topics } }) => ({
+    values: topics
+  }),
+  [getTopicsError]: (state, { payload }) => ({
+    ...state,
+    error: (payload && payload[0]) || 'An error occured. Please try again.'
```
Comment (manema) [inline src/redux/reducers/topicReducer.js:15]: If you will use that 'An error occured. Please try again.' in every reducer, it could be a constant. TIOLI

**Review by manema — APPROVED** [review]
Like a master. Getting worried about my job

File: src/redux/reducers/topicReducer.js (hunk 5 above)
Comment (afmicc) [reply src/redux/reducers/topicReducer.js:15]: Yeah, it's weird. I recently create a #22 that tries to solve that issue

**Pull request merged**